### PR TITLE
Arc 56

### PR DESCRIPTION
**Pull request by PatrickDinh**
Support ARC-56:
- deploy/edit application from App Lab given an ARC56 app spec
- call ABI methods from an ARC56 application
- decoded ABI information for historical app calls to an ARC56 application

DEMO:
https://github.com/user-attachments/assets/bf07b023-319b-4beb-8524-3c885e8d263e

### CODE REVIEW

File: src/features/transaction-wizard/components/method-call-transaction-builder.tsx [hunk 1]
```diff
@@ -70,46 +69,50 @@ export function MethodCallTransactionBuilder({
...
+      contract: _defaultValues?.appSpec,
     }
-  }, [_defaultValues?.appSpec, _defaultValues?.applicationId, _defaultValues?.method?.name, mode, transaction])
+  }, [_defaultValues?.appSpec, _defaultValues?.applicationId, _defaultValues?.methodDefinition?.name, mode, transaction])
 
   const [appId, setAppId] = useState<ApplicationId | undefined>(initialValues.applicationId)
   const [methodName, setMethodName] = useState<string | undefined>(initialValues.methodName)
-  const loadableMethodDefinitions = useLoadableAbiMethodDefinitions(initialValues.appSpec, appId)
+  const loadableArc56ContractWithMethodDefinitions = useLoadableArc56AppSpecWithMethodDefinitions(initialValues.contract, appId)
+  // TODO: PD - handle the struct in box contract in puya
```
Comment (PatrickDinh) [inline src/features/transaction-wizard/components/method-call-transaction-builder.tsx:87]: This will be done in another PR

File: src/features/common/utils.ts [hunk 2]
```diff
@@ -17,10 +18,23 @@ export function isArc32AppSpec(appSpec: Arc32AppSpec | Arc4AppSpec): appSpec is
...
     typeof appSpec === 'object' &&
+    !('arcs' in appSpec) &&
+    'methods' in appSpec &&
+    Array.isArray(appSpec.methods) &&
+    appSpec.methods.length > 0 &&
+    typeof appSpec.methods[0] === 'object'
+  )
+}
+
+export function isArc56AppSpec(appSpec: AppSpec): appSpec is Arc56Contract {
```
Comment (neilcampbell) [inline src/features/common/utils.ts:33]: "We should also check that 56 is included in the arcs array here" applies here too.
I'm wondering if we can share this check?

**Review by neilcampbell — APPROVED** [review]

File: src/features/applications/mappers/index.ts [hunk 3]
```diff
@@ -99,138 +101,164 @@ const getValue = (bytes: string) => {
...
+    const abiMethods = appSpec.methods.map((method) => {
+      return new algosdk.ABIMethod({
+        name: method.name,
+        desc: method.desc,
+        args: method.args,
+        returns: method.returns,
+      })
+    })
+
+    const arc32AppSpec = {
```
[the remark this answers is not in the document]
Comment (PatrickDinh) [reply src/features/applications/mappers/index.ts:121]: I'll keep this logic here so that we can utilised method `arc32ToArc56` from `utils-ts`

File: src/features/common/utils.ts (hunk 2 above)
Comment (PatrickDinh) [reply src/features/common/utils.ts:33]: good idea! I updated it to call `isXAppSpec` in `jsonAsXAppSpec` functions.
